File: CatalinaLesmes_GenerarTiempos.py
import time
 
#funcion recursiva de fibonacci

N = 35
def fibonacci(N):
	if(N<2):
		return N
	else:
		return fibonacci(N-1) + fibonacci(N-2)

#funcion que devuelve el tiempo que se demora la funcion fibonacci en calcular el n esimo numero

def get_time(N):
	t0 = time.time()
	fibonacci(N)
	t1 = time.time()-t0
	return t1
#imprime el resultado
for j in range(N+1):
	print j, get_time(j)

File: CatalinaLesmes_GenerarTiempos.cpp
#include <iostream>
#include <ctime>
using namespace std;
//funcion recursiva para obtener los valores de fibonacci pto 1

int fibonacci(int N){
	if(N < 2){
		return N;}
	else{
		return fibonacci(N-1) + fibonacci(N-2);}
	}
//funcion para retornar el numero de segundos que se demora la funcion de fibonacci en obtener el n-esimo numero
float get_time(int N){
	clock_t t;
	t = clock();
	fibonacci(N);
	t = clock() - t;
	float time = ((float)t)/CLOCKS_PER_SEC;
	return time;
	 }
//funcion que corre todo el codigo

int main() {
	int N;
	N = 35;
	fibonacci(N);
	get_time(N);
	cout << "Para N igual a 35 "<< " el tiempo es " << get_time(N) << endl;
	return 0;
}

